dashboard/dashapi/dashapi.go

```diff
@@ -236,6 +236,7 @@ type Crash struct {
 	Title       string
 	Corrupted   bool // report is corrupted (corrupted title, no stacks, etc)
 	Maintainers []string
+	Cc          []string
```

Comment:
Since this is part of stable API that should support forward/backward compatibility, it makes sense to think now how we will incorporate real names here. Adding 2 more parallel arrays here looks somewhat hacky. Maybe we should add a User type that contains email, real name, to/cc flags (or can we have more user types in future? @ebiggers) and then add array of these types here?
We will need the same info in few other structs in this file as well.

Reply:
Can you explain the forwards and backwards compatibility requirements?  New dashboard needs to work with old API clients, new API clients need to work with old dashboard, or both, or neither?  Is there any deprecation period?

I'm not sure what we should do.  One option would be to just format the email addresses as `Joe Example <joe@example.com>` like they would be in an actual email.  If we made the dashboard treat the name as optional, then old API clients would continue to work.  

Alternatively, to provide more future extensibility (without breaking old API clients) we could add an array like `Recipients []RecipientInfo` where

       type RecipientInfo {
                Email string
                RealName string
                RecipientType string
       }       

RecipientType could be "list", "maintainer", "reviewer", etc.

Old API clients would fill in `Maintainers`, new ones would use `Recipients`, and the dashboard would accept both.  (But again, to what extent is backwards compatibility really a requirement?)

Reply:
To some degree it depends on how much trouble we are ready to into to deploy. Ultimately we could stop entire syzbot and then bring up parts in particular order. But it's quite costly.
So preferably either compatibility both ways, or we could update dashboard first and then syz-ci/manager's, which won't be too hard (I will apply this PR locally and deploy to appengine and then we merge the change for syz-ci to pick up).
Deprecation period is short. Once all syzbot instances are upgrades, we can submit clean up changes or something.
For the datastore (entities.go) the story is a bit more problematic, we will need to deal with old data. Or do an explicit upgrade with some custom code, which is always somewhat painful.

String both name and email in the same string is interesting. We can first update the dashboard to understand both formats, and then we can update syz-ci/manager to include names.
But we will need to be careful to audit all uses in the dashboard to not mess things up (e.g. try to parse it only as email and fail spuriously, or break buganizer interface which may not expect names).